NOnion/Directory/TorDirectory.fs

```diff
@@ -208,8 +308,19 @@ type TorDirectory =
             return self.NetworkStatus
         }
 
+    /// WARNING: this function is scary and might throw KeyNotFoundException.
```

Comment:
Actually  it would  still be scary because it can return None

Reply:
no, you use tryFind but in case of None you throw failwith instead of KeyNotFoundException

Reply:
1. Failwith is worse because then I would need a generic try-with (Typed exception is fine but since you mentioned failwith)
2. It would still be scary because it can cause an exception

But what's the point of just changing the exception type?

Reply:
when I said you can use failwith, I was replying to your None-related comment about being scary, not about the place of code where you were catching the exception. The place where you are catching KeyNotFoundException now should rather be a match that does in the None case what it used to do in the catch. And that shouldn't be scary; what is scary is using a partial function

Reply:
I already changed this.

Reply:
https://github.com/nblockchain/NOnion/pull/30/commits/8a21725663fe5ed1ac2adc19727c3140d69313ca

Reply:
hmm.

Reply:
I think I didn't understand this before.

Reply:
Fixed in https://github.com/nblockchain/NOnion/pull/30/commits/242350dec2d54ff9d4382c9ec9c21602d1659c83.